server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -1377,19 +1376,52 @@ public GatedCloseable<IndexCommit> acquireSafeIndexCommit() throws EngineExcepti
         }
     }
 
+    /**
+     * Returns the lastest segmentInfos
+     */
+    public SegmentInfos getLatestSegmentInfos() {
+        return getEngine().getSegmentInfosSnapshot().get();
```

Comment:
I missed this before - getSegmentInfosSnapshot returns a `GatedCloseable`, we should use try-with-resources.

```java
        try(final GatedCloseable<SegmentInfos> snapshot = getEngine().getSegmentInfosSnapshot()) {
            return  snapshot.get();
        }
```

Reply:
+1 to the note above. How is this different from the existing _getSegmentInfosSnapshot_ ?

https://github.com/opensearch-project/OpenSearch/blob/04e43a708375bfa8ebb20fafad74679184b93ba9/server/src/main/java/org/opensearch/index/shard/IndexShard.java#L4029-L4031

Reply:
It is same as getSegmentInfosSnapshot(). So we can call that directly